### PR TITLE
Improved In App Browser and Chrome Custom Tabs

### PR DESCRIPTION
**Pull request by redreceipt**
## DESCRIPTION

Video demo: https://www.dropbox.com/s/zox1sc595qsdkr7/fix.mov?dl=0

Android Chrome Custom Tab:

Screen Shot 2019-09-04 at 6 44 10 PM

### What does this PR do, or why is it needed?

- Replaces our current (buggy) in app browser with the community supported version. This also gives us Chrome Custom Tabs.
- Replaces `UserBrowserWithCookie`

### What design trade-offs/decisions were made?

None.

### How do I test this PR?

Click any of the buttons the link externally.

### What automated tests did you write?

Added one to test Android.

## TODO:

- [x] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [x] Closes [#905]
- [x] No new warnings in tests, in storybook, and in-app
- [x] Upload GIF(s) of iOS and Android
- [x] Set a relevant reviewer

## REVIEW:

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

**Comment by redreceipt** [comment]
Android crashes. Blocked by #869, we need `androidx` (Material) support: https://github.com/material-components/material-components-android/issues/193

**Comment by richarddubay** [comment]
@redreceipt @conrad-vanl Because this needs Android X support, that means that we need to wait for RN 60 on this one, right?

**Comment by redreceipt** [comment]
Yea #869

**Comment by redreceipt** [comment]
Very close to getting cookies working but it doesn't look quite right coming through...

Screen Shot 2019-09-05 at 9 18 44 AM

**Comment by redreceipt** [comment]
Also, another note, don't think the `WebBrowserProvider` is working...but I don't think it was working before. I wanted to leave the code in there though so we can preserve what browser options are available and it's not really hurting anything. I just tried to implement it and got an error...

**Comment by redreceipt** [comment]
sike...headers is a iOS only feature 😞

**Comment by redreceipt** [comment]
ok this is ready to go now

File: packages/apolloschurchapp/src/ui/WebBrowser/index.js [hunk 1]
```diff
@@ -6,21 +6,32 @@ import { withTheme } from '@apollosproject/ui-kit';
 
 import Browser from './Browser';
 
-const { Provider: BaseProvider, Consumer } = createContext(Browser.openURL);
+const { Provider: BaseProvider, Consumer } = createContext(Browser.open);
 
 const Provider = withTheme(({ theme }) => ({
```
Comment (redreceipt) [inline packages/apolloschurchapp/src/ui/WebBrowser/index.js]:
```suggestion
// NOTE: don't think this is working...
const Provider = withTheme(({ theme }) => ({
```

File: packages/apolloschurchapp/src/ui/WebBrowser/Browser.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    const cookie = await getCookie();
+    // NOTE: iOS doesn't support headers, you may want to use src/user-web-view
+    const headers = { Cookie: cookie };
+    try {
+      if (await InAppBrowser.isAvailable()) {
+        const result = await InAppBrowser.open(url, {
+          headers,
+          ...options,
+        });
+        console.log(result);
```
Comment (vinnyjth) [inline packages/apolloschurchapp/src/ui/WebBrowser/Browser.js]:
```suggestion
```

File: packages/apolloschurchapp/src/ui/WebBrowser/Browser.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      query: GET_USER_COOKIE,
+    }
+  );
+  return rockToken;
+};
+
+const Browser = {
+  open: async (url, options) => {
+    const cookie = await getCookie();
+    // NOTE: iOS doesn't support headers, you may want to use src/user-web-view
```
Comment (vinnyjth) [inline packages/apolloschurchapp/src/ui/WebBrowser/Browser.js]:
```suggestion
    if (cookie) console.warn("iOS doesn't support headers, you may want to use src/user-web-view");
```

**Comment by vinnyjth** [comment]
One of my suggested changes must have broke your tests/linter @redreceipt. Once you fix I will merge this.

**Comment by redreceipt** [comment]
Something very strange is happening with tests here with the latest changes. Snapshots are changing on code that this one doesn't touch...

**Comment by redreceipt** [comment]
I guess it could be some dependency version that changed because I installed a new package, I'll update the snaps and then can you give them once over?

**Comment by redreceipt** [comment]
ok now I'm scratching my head as to what happened. `react-navigation` is causing `ui-onboarding` tests to fail here...
Screen Shot 2019-09-17 at 4 30 06 PM

**Pull request merged**